### ISSUE TITLE
[Submission] {22} params - {Archie Chaudhury}

### ISSUE THREAD

**Issue by Arch222** [anadim/AdderBoard#6]
### Author

@ArchChaudhury (Twitter)

### Unique Parameter Count

22

### Accuracy

100% on 10k random pairs

### Method

Hand-coded weights

### Architecture

- 1 decoder layer - hidden size = 3 - attention heads = 4 - KV heads = 1 - head dim = 2 - MLP hidden = 4 - vocab size = 10 (digit tokens only)

### Key Tricks
Fixed 3-channel state design: one large constant channel, one digit channel, one tiny decode-slope channel.
Tied embed/decode math: logits come from the same compact digit basis used in embeddings.
RoPE offset-targeted queries: queries are parameterized by 4 learned offsets so each head “locks” onto specific relative positions.
Shared single KV head expanded across 4 Q heads to cut parameters.
Procedural sparse output projection from attention context (only a few nonzero routes are used).
Carry logic encoded in SwiGLU thresholds (piecewise linear separators), not in Python state.
No learned positional embeddings; positional behavior is fixed via sinusoidal/RoPE operations.
Fixed-format prompt and fixed n_digits=10 calibration to keep the constructive weight solution stable.

### Link to Code

https://github.com/Arch222/Addition_Final

### Verification Output

```shell
Model: Handwritten Compressed Transformer Adder
Unique params (reported): 22
Accuracy: 10010 / 10010 = 100.00%
Time: 93.9s (107 additions/sec)
Status: QUALIFIED (>= 99%)
```
